source/common/http/header_map_impl.cc

```diff
@@ -150,6 +150,46 @@ void HeaderString::append(const char* data, uint32_t size) {
   string_length_ += size;
 }
 
+void HeaderString::rtrim() {
```

Comment:
I am not sure why and how rtrim() is invoked only with  `Type::Inline` implementation. 
It might be true in #10886 but not here.
Any advices?

Reply:
There are 2 operations that are done on HeaderString: Construction and append.  Construction of an empty HeaderString results in an Inline headerstring.  append converts to inline if not already inline.

Can you provide info on where rtrim is called on non Inline headerstrings?